provider/servicefabric/servicefabric.go

```diff
@@ -0,0 +1,420 @@
+	for _, service := range services {
+		if p.hasServiceLabel(service, key) {
+			srvWithLabel = append(srvWithLabel, service)
+		}
+	}
+	return srvWithLabel
+}
+
+func (p *Provider) getServicesWithLabelValue(services []ServiceItemExtended, key, expectedValue string) []ServiceItemExtended {
+	srvWithLabel := []ServiceItemExtended{}
+	for _, service := range services {
+		value, exists := service.Labels[key]
+		if exists && value == expectedValue {
+			srvWithLabel = append(srvWithLabel, service)
+		}
+	}
+	return srvWithLabel
+}
+
+func (p *Provider) getServiceLabelValueWithDefault(service ServiceItemExtended, key, defaultValue string) string {
+	value, exists := service.Labels[key]
+
+	if !exists {
+		return defaultValue
+	}
+
+	return value
+}
+
+func (p *Provider) getServiceLabelsWithPrefix(service ServiceItemExtended, prefix string) map[string]string {
+	results := make(map[string]string)
+	for k, v := range service.Labels {
+		if strings.HasPrefix(k, prefix) {
+			results[k] = v
+		}
+	}
+	return results
+}
+
+func (p *Provider) isPrimary(instance sfsdk.ReplicaInstance) bool {
+	_, data := instance.GetReplicaData()
+	primaryString := "Primary"
+	if data.ReplicaRole == primaryString {
+		return true
+	}
+	return false
+}
+
+func (p *Provider) doesAppParamContain(app sfsdk.ApplicationItem, key, shouldContain string) bool {
+	value := p.getApplicationParameter(app, key)
+	if strings.Contains(value, shouldContain) {
+		return true
+	}
+	return false
+}
+
+func (p *Provider) getApplicationParameter(app sfsdk.ApplicationItem, key string) string {
+	for _, param := range app.Parameters {
+		if param.Key == key {
+			return param.Value
+		}
+	}
+	log.Errorf("Parameter %s doesn't exist in app %s", key, app.Name)
+	return ""
+}
+
+func (p *Provider) getDefaultEndpoint(instance sfsdk.ReplicaInstance) string {
+	id, data := instance.GetReplicaData()
+	endpoint, err := getDefaultEndpoint(data.Address)
+	if err != nil {
+		log.Warnf("No default endpoint for replica %s in service %s endpointData: %s", id, data.Address)
+		return ""
+	}
+	return endpoint
+}
+
+func (p *Provider) getNamedEndpoint(instance sfsdk.ReplicaInstance, endpointName string) string {
+	id, data := instance.GetReplicaData()
+	endpoint, err := getNamedEndpoint(data.Address, endpointName)
+	if err != nil {
+		log.Warnf("No names endpoint of %s for replica %s in endpointData: %s", endpointName, id, data.Address)
+		return ""
+	}
+	return endpoint
+}
+
+func isHealthy(instanceData sfsdk.ReplicaItemBase) bool {
+	if instanceData.ReplicaStatus == "Ready" || instanceData.HealthState != "Error" {
+		return true
+	}
+	return false
+}
+
+func hasHTTPEndpoint(instanceData sfsdk.ReplicaItemBase) bool {
+	_, err := getDefaultEndpoint(instanceData.Address)
+	return err == nil
+}
+func decodeEndpointData(endpointData string) (map[string]string, error) {
+	var endpointsMap map[string]map[string]string
+
+	if endpointData == "" {
+		return nil, errors.New("Endpoint data is empty")
```

Comment:
Could you use a lower-case first letter?

https://github.com/golang/go/wiki/CodeReviewComments#error-strings